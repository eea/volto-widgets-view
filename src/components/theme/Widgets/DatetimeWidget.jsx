import React from 'react';
import cx from 'classnames';
import moment from 'moment';

// import 'moment/locale/en-gb';
// import { useIntl } from 'react-intl';

export const DatetimeWidget = ({ children, className }) => {
  // const intl = useIntl();
  // moment.locale(intl.locale);
  const datetime = moment(children).format('lll');
  return (
    <span className={cx(className, 'datetime', 'widget')}>{datetime}</span>
  );
};
