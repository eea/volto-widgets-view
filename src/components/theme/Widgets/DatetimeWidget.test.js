import React from 'react';
import renderer from 'react-test-renderer';
import { DatetimeWidget } from './DatetimeWidget';

describe('DatetimeWidget', () => {
  it('renders a date view widget component', () => {
    const component = renderer.create(
      <DatetimeWidget className="metadata">2020-08-04T09:00:00</DatetimeWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
