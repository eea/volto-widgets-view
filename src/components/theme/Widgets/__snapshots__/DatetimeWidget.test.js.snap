// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatetimeWidget renders a date view widget component 1`] = `
<span
  className="metadata datetime widget"
>
  Aug 4, 2020 9:00 AM
</span>
`;
